cluster/addons/iso-client/opaque/opaque.go

```diff
@@ -61,20 +35,36 @@ func getNode() (string, error) {
 }
 
 // path for opaque resources
-func (opaque *OpaqueIntegerResourceAdvertiser) generateOpaqueResourcePath() string {
-	return fmt.Sprintf("/status/capacity/pod.alpha.kubernetes.io~1opaque-int-resource-%s", opaque.Name)
+func generateOpaqueResourcePath(name string) string {
+	return fmt.Sprintf("/status/capacity/pod.alpha.kubernetes.io~1opaque-int-resource-%s", name)
```

Comment:
It would be preferable to turn the name into an OpaqueIntResource name using the provided helpers and then building up the path using the result. There's a helper here: https://github.com/kubernetes/client-go/blob/623dbc3dc4e22ddd57b7d1d73b9afa9c9448e8f5/pkg/api/helpers.go#L148

If we leave it as-is, then we have to remember to update this string literal when moving OIR from alpha to beta and again during beta to GA.